src/async-map.mjs

```diff
@@ -13,74 +13,76 @@ class ParallelMapper {
-              // internally, as if they were requests. If there is a request in the queue, they
-              // begin to generate the next item as soon as the last one is finished, whether or not
-              // anybody has tried to consume it from ParallelMapper yet. This is what gives us
-              // us parallelism. Without the ability to read ahead in the source, we would have
-              // to wait until we had produced one mapped item until our consumer could call our
-              // next method again triggering another source fetch. This would result in completely
-              // serial behavior.
-              buffer.push(this.next())
-            }
-          }
+      return bufferValue
+    }
 
-          let mappedValue
-          if (!item.done) {
-            mappedValue = await mapPromise
+    const promise = iter.next()
+      .then(async item => {
+        // We are guaranteed by the async generator impl that this then block
+        // will be executed in order
+        const { previousUnresolvedPromise } = this
+
+        this.previousUnresolvedPromise = promise
+
+        let mapPromise
+        if (item.done) {
+          this.done = true
+        } else {
+          mapPromise = mapFn(item.value)
+
+          if (!buffer.isFull()) {
+            // Go on filling up the buffer in the background
+            // Note that we don't await the last generator item before requesting the next one.
+            // This makes the code "eager". It works because async generators queue `next` calls
+            // internally, as if they were requests. If there is a request in the queue, they
+            // begin to generate the next item as soon as the last one is finished, whether or not
+            // anybody has tried to consume it from ParallelMapper yet. This is what gives us
+            // us parallelism. Without the ability to read ahead in the source, we would have
+            // to wait until we had produced one mapped item until our consumer could call our
+            // next method again triggering another source fetch. This would result in completely
+            // serial behavior.
+            buffer.push(this.next())
           }
+        }
 
-          if (previousUnresolvedPromise) {
-            // Ensure we don't emit parallelized items out of order
-            await previousUnresolvedPromise
-          }
+        let mappedValue
+        if (!item.done) {
+          mappedValue = await mapPromise
+        }
 
-          if (item.done) {
-            return item
-          } else {
-            return { value: { value: item.value, mappedValue }, done: false }
-          }
-        })
-    }
+        if (previousUnresolvedPromise) {
+          // Ensure we don't emit parallelized items out of order
+          await previousUnresolvedPromise
+        }
+
+        if (item.done) {
+          return item
+        } else {
+          return { value: { value: item.value, mappedValue }, done: false }
```

Comment:
I forgot to touch this. There's no longer any need to emit and object including the value.

Reply:
I am not sure I understand this

Reply:
Like:
```js
return { value:  mappedValue , done: false }
```
and 
```js
  for await (const value of runner) {
    yield value
  }
```
Am I correct?

Reply:
fixed